server/routes/invitation.js

```diff
@@ -26,4 +27,37 @@ router.get("/user/:id",
     }
 );
 
+// Returns contacts that the current user accepted invitations from
+router.get("/user/:id/contacts",
+    passport.authenticate('jwt', { session: false }),
+    function(req, res, next) {
+        const {id} = req.params;
+        var objId = mongoose.Types.ObjectId;
+
+        Invitation.find({"to_user": new objId(id), "approved": true },
+            function (err, invitations) {
+                if (err) {
+                    return handleError(err);
+                }
+                if (invitations && invitations.length) {
+                    let contacts = [];
+                    invitations.forEach((invite, idx) => {
+                        User.findById(invite.from_user.toString(), function(err, user) {
```

Comment:
i'm wondering if you can use Mongo `.populate` method to have mongo populate all this data for you all at once 

https://mongoosejs.com/docs/populate.html

aggregate the data, combining userId on Invitation to grab all the data at once

Reply:
in one earlier version I was using aggregate but I took it out. for some reason I felt like I was getting a lot of duplicate data but it could have been my just using duplicate test data. I will go back to the aggregate method and test it again (for accuracy).

Reply:
hi, I updated the PR. I reverted to older code where I was using aggregate with $look. I tried to look up which is preferred, populate vs aggregate and it looked like aggregate might have better performance.

* https://github.com/Automattic/mongoose/issues/3683
* https://stackoverflow.com/questions/55575806/mongoose-populate-vs-aggregate

Reply:
aewsome looks fantastic!